pandas/tests/frame/test_analytics.py

```diff
@@ -815,59 +826,54 @@ def wrapper(x):
                                                      skipna_alternative)
             result0 = f(axis=0, skipna=False)
             result1 = f(axis=1, skipna=False)
-            tm.assert_series_equal(result0, frame.apply(wrapper),
+            tm.assert_series_equal(result0, main_frame.apply(wrapper),
                                    check_dtype=check_dtype,
                                    check_less_precise=check_less_precise)
             # HACK: win32
-            tm.assert_series_equal(result1, frame.apply(wrapper, axis=1),
+            tm.assert_series_equal(result1, main_frame.apply(wrapper, axis=1),
                                    check_dtype=False,
                                    check_less_precise=check_less_precise)
         else:
             skipna_wrapper = alternative
-            wrapper = alternative
 
         result0 = f(axis=0)
         result1 = f(axis=1)
-        tm.assert_series_equal(result0, frame.apply(skipna_wrapper),
+        tm.assert_series_equal(result0, main_frame.apply(skipna_wrapper),
                                check_dtype=check_dtype,
                                check_less_precise=check_less_precise)
         if name in ['sum', 'prod']:
-            exp = frame.apply(skipna_wrapper, axis=1)
+            exp = main_frame.apply(skipna_wrapper, axis=1)
             tm.assert_series_equal(result1, exp, check_dtype=False,
                                    check_less_precise=check_less_precise)
 
         # check dtypes
         if check_dtype:
-            lcd_dtype = frame.values.dtype
+            lcd_dtype = main_frame.values.dtype
             assert lcd_dtype == result0.dtype
             assert lcd_dtype == result1.dtype
 
-        # result = f(axis=1)
-        # comp = frame.apply(alternative, axis=1).reindex(result.index)
-        # assert_series_equal(result, comp)
-
         # bad axis
         tm.assert_raises_regex(ValueError, 'No axis named 2', f, axis=2)
         # make sure works on mixed-type frame
-        getattr(self.mixed_frame, name)(axis=0)
-        getattr(self.mixed_frame, name)(axis=1)
+        getattr(float_string_frame, name)(axis=0)
```

Comment:
From here on I'm thinking these tests should live in a separate class method, or be broken out some other way.